_posts/2019-07-30-censored-data/censored_data_tfp.Rmd

```diff
@@ -0,0 +1,724 @@
+---
+title: "Modeling censored data with tfprobability"
+description: > 
+ In this post we use tfprobability, the R interface to TensorFlow Probability, to model censored data. Again, the exposition is inspired by the treatment of this topic in Richard McElreath's Statistical Rethinking. Instead of cute cats though, we model immaterial entities from the cold world of technology: This post explores durations of CRAN package checks, a dataset that comes with Max Kuhn's parsnip.
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydana2019tfpcensored
+date: 07-30-2019
+categories:
+  - Probability and statistics
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/thumb_cropped.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+Nothing's ever perfect, and data isn't either. One type of "imperfection" is _missing data_, where some features are unobserved for some subjects. (A topic for another post.) Another is _censored data_, where an event whose characteristics we want to measure does not occur in the observation interval. The example in Richard McElreath's _Statistical Rethinking_ is time to adoption of cats in an animal shelter. If we fix an interval and observe wait times for those cats that actually _did_ get adopted, our estimate will end up too optimistic: We don't take into account those cats who weren't adopted during this interval and thus, would have contributed wait times of length longer than the complete interval.
+
+In this post, we use a slightly less emotional example which nonetheless may be of interest, especially to R package developers: time to completion of `R CMD check`, collected from CRAN and provided by the `parsnip` package as `check_times`. Here, the censored portion are those checks that errored out for whatever reason, i.e., for whom the check did not complete.
+
+Why do we care about the censored portion, the entities for whom the event of interest did not occur in the period of observation? In the cat adoption scenario, this is pretty obvious: We want to be able to get a realistic estimate for any unknown cat, not just those cats that will turn out to be "lucky". How about `check_times`? Well, if your submission is one of those that errored out, you still care about how long you wait, so even though their percentage is low (< 1%) we don't want to simply exclude them. Also, there is the possibility that the failing ones would have taken longer, had they run to completion, dut to some intrinsic difference between both groups. Conversely, if failures were random, the longer-running checks would have a greater chance to get hit by an error. So here too, exluding the censored data may result in bias.
```

Comment:
I would open this with the question: "Why do we care about the censored portion?" Then describe the cat scenario (as you do now), _then_ go into the check times/entities for which event of interest didn't occur.

Reply:
Hm, I think that is the order I'm following?
I mean, 

"the entities for whom the event of interest did not occur in the period of observation?"

is really nothing than an "extended paraphrase"?

Reply:
Yeah — it's extended enough that it makes it a bit confusing to have the cats in the middle, and then go back to it.

Reply:
Yeah you're right, I'm probably overdoing it rephrasing things times & again ... I  think I'll just remove the paraphrase completely, right?